src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -325,6 +350,19 @@ void remove(int filteredIndex) throws IllegalValueException {
             }
         }
 
+        void removeAll(Collection<E> toRemove) {
+            for (E item : toRemove) {
+                if (filteredList.contains(item)) {
```

Comment:
The predicate used in `removeAll`, and the predicate we should use in `hide-finished` command is completely opposite.